### PR TITLE
fix: Prevent `Fallback forward departments` feature to go on loop when configured as circular chain

### PR DESCRIPTION
**Pull request by KevLehman**
[SUP-449](https://rocketchat.atlassian.net/browse/SUP-449)

## Further comments

[SUP-449]: https://rocketchat.atlassian.net/browse/SUP-449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lmauromb — APPROVED** [review]

**Review by lmauromb — APPROVED** [review]

File: apps/meteor/ee/app/livechat-enterprise/server/hooks/onTransferFailure.ts [hunk 1]
```diff
@@ -12,27 +12,17 @@ const onTransferFailure = async (
 	{
 		guest,
 		transferData,
+		department,
 	}: {
 		guest: ILivechatVisitor;
 		transferData: TransferData;
+		department: AtLeast<ILivechatDepartment, '_id' | 'fallbackForwardDepartment'>;
```
Comment (MarcosSpessatto) [inline apps/meteor/ee/app/livechat-enterprise/server/hooks/onTransferFailure.ts:19]: Isn't `name` required as well?

File: apps/meteor/lib/callbacks.ts [hunk 2]
```diff
@@ -128,7 +128,11 @@ type ChainedCallbackSignatures = {
 	'livechat:afterOnHoldChatResumed': (room: Pick<IOmnichannelRoom, '_id'>) => Pick<IOmnichannelRoom, '_id'>;
 	'livechat:onTransferFailure': (
 		room: IRoom,
-		params: { guest: ILivechatVisitor; transferData: TransferData },
+		params: {
+			guest: ILivechatVisitor;
+			transferData: TransferData;
+			department: AtLeast<ILivechatDepartmentRecord, '_id' | 'fallbackForwardDepartment'>;
```
Comment (MarcosSpessatto) [inline apps/meteor/lib/callbacks.ts:134]: Isn't `name` required as well?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Would it be possible to add some test cases for the maximum number of depth?

File: apps/meteor/ee/app/livechat-enterprise/server/hooks/onTransferFailure.ts (hunk 1 above)
Comment (KevLehman) [inline apps/meteor/ee/app/livechat-enterprise/server/hooks/onTransferFailure.ts:19]:
```suggestion
		department: AtLeast<ILivechatDepartment, '_id' | 'fallbackForwardDepartment' | 'name'>;
```

File: apps/meteor/lib/callbacks.ts (hunk 2 above)
Comment (KevLehman) [inline apps/meteor/lib/callbacks.ts:134]:
```suggestion
			department: AtLeast<ILivechatDepartmentRecord, '_id' | 'fallbackForwardDepartment' | 'name'>;
```

**Comment by KevLehman** [comment]
Tests are definitely possible. The test would take some time however to create the number of departments required for properly testing depth, so at the end I didn't do it 🤔 

I can include it on a separate PR too.

**Review by lmauromb — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**